packages/devtools/lib/src/memory/memory.dart

```diff
@@ -362,38 +682,357 @@ class MemoryScreen extends Screen with SetStateMixin {
+        memoryPath.addPathsLinks(history, _handleHistoryClicks);
+
+        history..clazz('history-navigation', removeOthers: true);
+      }
+    }
+  }
+
+  void _handleHistoryClicks(CoreElement element) {
+    // Handle clicking in the history links.
+    if (element.tag == 'SPAN' && element.hasClass('history-link')) {
+      final attrs = element.attributes;
+
+      final int dataIndex = int.parse(attrs[NavigationState.dataIndex]);
+
+      final String dataClass = attrs[NavigationState.dataClass];
+      final String dataField = attrs.containsKey(NavigationState.dataField)
+          ? attrs[NavigationState.dataField]
+          : '';
+      final int dataHashCode = attrs.containsKey(NavigationState.dataHashCode)
+          ? int.parse(attrs[NavigationState.dataHashCode])
+          : null;
+
+      final NavigationState state = memoryPath.get(dataIndex);
+
+      // The clicked link's attributes and real NavigationState should match.
+      assert(dataClass == state.className &&
+          dataField == state.field &&
+          dataHashCode == state.instanceHashCode);
+
+      // Prune remove this state and to the end as well.
+      memoryPath.remove(state);
+
+      if (state.isClass) {
+        _selectClass(state.className);
+      } else if (state.isInstance) {
+        selectClassInstance(state.className, state.instanceHashCode);
+      } else if (state.isInbound) {
+        // Same as selecting the class instance but record the field, don't
+        // need to match following the refs.
+        memoryPath.fieldReference = state.field;
+        selectClassInstance(state.className, state.instanceHashCode);
+      } else {
+        assert(false, 'Unknown NavigationState');
+      }
+
+      history.clear();
+      Timer(const Duration(milliseconds: 100), () {
+        if (!memoryPath.isLastInBound) {
+          history..clazz('history-navigation-hidden', removeOthers: true);
+        } else {
+          memoryPath.addPathsLinks(history, _handleHistoryClicks);
+        }
+      });
+    }
+  }
+
+  // TD element used to simulate hover state when hover card is visible. When
+  // not null the mouse is actively in the hover card.
+  CoreElement _tdCellHover;
+
+  // InstanceSummary of the visible hover card.
+  HoverCellData<InstanceSummary> _currentHoverSummary;
+
+  // This is the listener for the hover card (hoverPopup's) onMouseOver, it's
+  // designed to keep the hover state (background-color for the TD same as the
+  // CSS :hover) as the mouse slides to the hover card. It gives the appearance
+  // that hover is still active in the TD.
+  void _mouseInHover(html.MouseEvent evt) {
+    CoreElement cell = _currentHoverSummary?.cell;
+
+    if (cell != null) _tdCellHover = cell;
 
-        // Allow inspection of the memory object.
-        memoryDataView.showFields(instance != null ? instance.fields : []);
+    // Simulate the :hover when the mouse in hover card.
+    _tdCellHover?.clazz('allocation-hover', removeOthers: true);
```

Comment:
add both classes with the same line?

Reply:
Not sure how to clazz will not allow spaces between clazz names.